enhancements/cluster-logging/performance-tuning.md

```diff
@@ -0,0 +1,241 @@
+---
+title: performance-tuning
+authors:
+  - "@alanconway"
+reviewers:
+  - "@jcantrill"
+approvers:
+  - "@jcantrill"
+api-approvers:
+  - "@jcantrill"
+creation-date: 2023-12-15
+last-updated: 2023-12-15
+tracking-link:
+  - https://issues.redhat.com/browse/OBSDA-549
+see-also:
+replaces:
+superseded-by:
+---
+
+# Performance tuning
+
+## Summary
+
+A _perfance tuning_ API to control performance, reliability and special protocol features of an output,
+without exposing the complexity of the underlying collector configuration.
+
+**Note**
+- Only vector will be supported, there are no plans to back-port to fluentd.
+- Existing `output[].limits` allows rate limiting, that is separate from this proposal.
+
+## Motivation
+
+Performance tuning in the underlying collector configuration is complex and error prone.
+Achieving a performance goal may involve configuring multiple parts of the delivery pipeline consistently.
+This would be difficult and error prone for users, but it can be safely automated by our configuration generator.
+
+### User Stories
+#### As a cluster logging administrator I want to minimize log loss.
+
+``` yaml
+outputs:
+ new - name: minimise-log-loss
+   tuning:
+     delivery: safe
+```
+
+#### As a cluster logging administrator I want to maximize throughput.
+
+``` yaml
+outputs:
+  - name: maximise-throughput
+    tuning:
+      delivery: fast
+```
+
+#### As a cluster logging administrator I want to tweak features of a specific output.
+
+``` yaml
+outputs:
+  - name: detailed-tweaks
+    tuning:
```

Comment:
This does not seem to reflect the rest of the document but adding comment here.

Maybe if the delivery implies memory or disk buffer then I still believe we need some way to further tune events and disk size. I would expect then we might provide something like:

```
tuning:
  delivery: fast|safe
  compression:
  request:
    maxDelay
  safe:
    maxEvents:
  fast:
    maxSize:    #this is bytes but could be '1G' which we convert ?
```

* I have comments down below regarding maxSize which I think we can calculate or make some assumptions and not allow it to be tweaked.
* I think we have to use vector term 'request' otherwise in future if we need to expose more, I don't see how the other parameters translate to 'reconnect'